PWGHF/HFL/Tasks/taskElectronWeakBoson.cxx

```diff
@@ -0,0 +1,248 @@
+    // LOGF(info, "Number of tracks: %d", tracks.size());
+    // LOGF(info, "Number of clusters: %d", clusters.size());
+    // LOGF(info, "Number of clusters: %d", emcClusters.size());
+
+    histos.fill(HIST("ZvtxHistogram"), collision.posZ());
+
+    for (const auto& track : tracks) {
+
+      histos.fill(HIST("etaHistogram"), track.eta());
+      histos.fill(HIST("ITS_Chi2_Hist"), track.itsChi2NCl());
+      histos.fill(HIST("TPC_Chi2_Hist"), track.tpcChi2NCl());
+      histos.fill(HIST("TPC_NCls_Hist"), track.tpcNClsFound());
+      histos.fill(HIST("ITS_NCls_Hist"), track.itsNCls());
+      histos.fill(HIST("TPC_NClsCrossedRows_Hist"), track.tpcNClsCrossedRows());
+
+      if (std::abs(track.eta()) > etaup)
+        continue;
+      if (track.tpcNClsCrossedRows() < tpcNclCr_cut)
+        continue;
+      if (std::abs(track.dcaXY()) > dcaxy_cut)
+        continue;
+      if (track.itsChi2NCl() > itschi2_cut)
+        continue;
+      if (track.tpcChi2NCl() > tpcchi2_cut)
+        continue;
+      if (track.tpcNClsFound() < tpcNcl_cut)
+        continue;
+      if (track.itsNCls() < itsNcl_cut)
+        continue;
+      if (track.pt() < mimpT_cut)
+        continue;
+      histos.fill(HIST("ptHistogram"), track.pt());
+      histos.fill(HIST("TPCElHistogram"), track.p(), track.tpcNSigmaEl());
+
+      // track - match
+
+      if (emcClusters.size() < 1)
+        continue;
+      if (track.phi() < 1.39 || track.phi() > 3.15)
+        continue;
+      auto tracksofcluster = matchedtracks.sliceBy(perClusterMatchedTracks, track.globalIndex());
+
+      // LOGF(info, "Number of matched track: %d", tracksofcluster.size());
+
+      double Rmim = 999.9;
+      double dPhi_mim = 999.9;
+      double dEta_mim = 999.9;
+
+      if (tracksofcluster.size() > 0) {
+        int nmatch = 0;
+        for (const auto& match : tracksofcluster) {
+          if (match.emcalcluster_as<SelectedClusters>().time() < minTime || match.emcalcluster_as<SelectedClusters>().time() > maxTime)
+            continue;
+          if (match.emcalcluster_as<SelectedClusters>().m02() < minM02 || match.emcalcluster_as<SelectedClusters>().m02() > maxM02)
+            continue;
+
+          double emc_m20 = match.emcalcluster_as<SelectedClusters>().m20();
+          double emc_m02 = match.emcalcluster_as<SelectedClusters>().m02();
+          double emc_energy = match.emcalcluster_as<SelectedClusters>().energy();
+          double emc_phi = match.emcalcluster_as<SelectedClusters>().phi();
+          double emc_eta = match.emcalcluster_as<SelectedClusters>().eta();
+          double emc_time = match.emcalcluster_as<SelectedClusters>().time();
+          // LOG(info) << "tr phi0 = " << match.track_as<TrackEle>().phi();
+          // LOG(info) << "tr phi1 = " << track.phi();
+          // LOG(info) << "emc phi = " << emc_phi;
+          if (nmatch == 0) {
+            double dPhi = match.track_as<TrackEle>().phi() - emc_phi;
```

Comment:
You should constrain `dPhi` to be between -π and +π for distance calculations. See `RecoDecay::constrainAngle`.